src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -134,16 +134,19 @@ protected virtual ParameterDescriptor CreateTriggerParameter(BindingMetadata tri
             switch (triggerMetadata.Type)
             {
                 case BindingType.QueueTrigger:
-                    triggerParameter = ParseQueueTrigger((QueueBindingMetadata)triggerMetadata, parameterType ?? typeof(string));
+                    parameterType = DefaultParameterType(parameterType, triggerMetadata.DataType);
```

Comment:
These changes allow the function do declare a byte[] etc. trigger value, so their scripts receive a byte[]. Note that for the bindings that are binding to Stream we don't need to do anything here - the conversion happens later when we read the value from the stream.